client/src/components/Profile/ContactsCard.tsx

```diff
@@ -1,245 +1,159 @@
-import * as React from 'react';
-import isEqual from 'lodash/isEqual';
-import { List, Typography, Input, Form } from 'antd';
-import CommonCard from './CommonCard';
-import { Contacts } from 'common/models/profile';
-import { ConfigurableProfilePermissions } from 'common/models/profile';
-import { ChangedPermissionsSettings } from 'pages/profile';
-import { CheckboxChangeEvent } from 'antd/lib/checkbox';
-
-const { Text } = Typography;
-
+import React, { useEffect, useMemo, useState } from 'react';
+import { List, Typography } from 'antd';
 import { ContactsOutlined } from '@ant-design/icons';
+import { isEqual } from 'lodash';
+import { UpdateProfileInfoDto } from 'api';
+import { Contacts } from 'common/models/profile';
 import { NotificationChannel } from 'modules/Notifications/services/notifications';
 import { EmailConfirmation } from './EmailConfirmation';
-import { Rule } from 'antd/lib/form';
+import CommonCardWithSettingsModal from './CommonCardWithSettingsModal';
+import { Contact, ContactsKeys } from 'services/user';
+import ContactsCardForm from './ContactsCardForm';
+
+const { Text } = Typography;
 
-type ProfileEntryChangeCallback = (event: { target: { value: string } }, path: string) => void;
+type Connections = Partial<
+  Record<
+    NotificationChannel,
+    | {
+        value: string;
+        enabled: boolean;
+        lastLinkSentAt?: string;
+      }
+    | undefined
+  >
+>;
```

Comment:
Minor one: I know that this is auto-format issue but to avoid this weird union type at line 18 I would suggest to replace it with something like 
```suggestion
type ConnectionValue = {
  value: string;
  enabled: boolean;
  lastLinkSentAt?: string;
};

type Connections = Partial<Record<NotificationChannel, ConnectionValue | undefined>>;
```

Reply:
sure)